Add support for 'cache-features' option in layers.

Algunenano opened the pull request.

BHare1985 commented:
I have a forked version of grainstore where I added cache-features in a different manner. I am showcasing that commit just for more context on another way it could be done;

https://github.com/BHare1985/grainstore/commit/fc284f7720e1cfd84959ecd43088408d198d485f

This would allow me to set any properties of the layer, including minzoom and maxzoom and anything else the caller would like, rather than have specific one-off code on a case by case basis.

Since my fork the getLayerProperties had been added ( https://github.com/Algunenano/grainstore/commit/407d55aadceee91919684b0606f85c7cf075a614#diff-5828130bd2086f5e8ac81d03cead23b0 ) , so keep that in mind.

Algunenano commented:
Upon reviewing it, I see that properties is supposed to receive an array per each property (one value per layer) so I'll rework the PR to work the same way as currently it's expecting just true/false for all layers.

Algunenano commented:
I've updated the PR to add support to both passing an array (to set a property per layer) or just an object. As I wasn't sure about supporting just anything (as @BHare1985 's solution), I've added the properties currently used in Mapnik's parser: `minimum-scale-denominator`, `maximum-scale-denominator`, `queryable`, `clear-label-cache`, `cache-features`, `group-by`, `buffer-size` and `maximum-extent`.
`minzoom` and `maxzoom` are treated differently so it doesn't break compatibility.
